WebGPU API docs (#24202)
mdn/content: files/en-us/web/api/gpudevice/createbuffer/index.md

status:
  - experimental
browser-compat: api.GPUDevice.createBuffer
---

{{APIRef("WebGPU API")}}{{SeeCompatTable}}

The **`createBuffer()`** method of the
{{domxref("GPUDevice")}} interface creates a {{domxref("GPUBuffer")}} in which to store raw data to use in GPU operations.

## Syntax

```js-nolint
createBuffer(descriptor)
```

### Parameters

- `descriptor`

  - : An object containing the following properties:

    - `label` {{optional_inline}}
      - : A string providing a label that can be used to identify the object, for example in {{domxref("GPUError")}} messages or console warnings.
    - `mappedAtCreation` {{optional_inline}}

      - : A boolean. If set to `true`, the buffer will be mapped upon creation, meaning that you can set the values inside the buffer immediately by calling {{domxref("GPUBuffer.getMappedRange()")}}. The default value is `false`.

        Note that it is valid to set `mappedAtCreation: true` so you can set the buffer's initial data, even if the `GPUBufferUsage.MAP_READ` or `GPUBufferUsage.MAP_WRITE` usage flags are not set.

    - `size`
      - : A number representing the size of the buffer, in bytes.
    - `usage`

>       - : The {{glossary("bitwise flags")}} representing the allowed usages for the `GPUBuffer`. The possible values can be found in the [`GPUBuffer.usage` value table](/en-US/docs/Web/API/GPUBuffer/usage#value).

Reviewer: ```suggestion
      - : The {{glossary("Bitwise flags", "bitwise flags")}} representing the allowed usages for the `GPUBuffer`. The possible values are in the [`GPUBuffer.usage` value table](/en-US/docs/Web/API/GPUBuffer/usage#value).
```

---
title: GPUDevice.createBuffer()
slug: Web/API/GPUDevice/createBuffer
page-type: web-api-instance-method
status:
  - experimental
browser-compat: api.GPUDevice.createBuffer
---

{{APIRef("WebGPU API")}}{{SeeCompatTable}}

The **`createBuffer()`** method of the
{{domxref("GPUDevice")}} interface creates a {{domxref("GPUBuffer")}} in which to store raw data to use in GPU operations.

## Syntax

```js-nolint
createBuffer(descriptor)
```

### Parameters

- `descriptor`

  - : An object containing the following properties:

    - `label` {{optional_inline}}
      - : A string providing a label that can be used to identify the object, for example in {{domxref("GPUError")}} messages or console warnings.
    - `mappedAtCreation` {{optional_inline}}

>       - : A boolean. If set to `true`, the buffer will be mapped upon creation, meaning that JavaScript can access the values inside the buffer by calling {{domxref("GPUBuffer.getMappedRange()")}} immediately. The default value is `false`.

Author: Noted. I ended up fixing this in a slightly different way as I felt that it read a bit awkwardly. Is this OK?

> If set to `true`, the buffer will be mapped upon creation, meaning that you can set the values inside the buffer immediately by calling {{domxref("GPUBuffer.getMappedRange()")}}.

>       - : A boolean. If set to `true`, the buffer will be mapped upon creation, meaning that JavaScript can access the values inside the buffer by calling {{domxref("GPUBuffer.getMappedRange()")}} immediately. The default value is `false`.

Reviewer 2: Perfect! Thanks for refining it!
